bootstrap_salt/fab_tasks.py

```diff
@@ -100,211 +105,121 @@ def application(application_name):
     env.application = str(application_name).lower()
 
 
-@task
-def setup(salt_version='v2014.7.5'):
+def get_kms_key_id():
     """
-    Setup the salt master and minions
-
-    Call install_master and install_minions to setup salt
+    Returns an id for a KMS master key with an alias of:
+    application-environment. If no key exists it will create a
+    new one.
     """
-    install_master(salt_version)
-    install_minions(salt_version)
+    alias = "{0}-{1}".format(env.application, env.environment)
+    kms = get_connection(KMS)
+    key_id = kms.get_key_id(alias)
+    if not key_id:
+        key_id = kms.create_key(alias)
+    return key_id
```

Comment:
They will share a master key yes. However, each time you create a stack you get a new data key. I figure that's ok because it's still unique between envs. The reason for this is I didn't want to change how we get the stack name.

The S3 bucket is unique to each stack.